### ISSUE TITLE
After several iterations the error:  "input and filter must have the same depth" is arose

### ISSUE THREAD

**Issue by PEYMAN333** [tensorflow/tensorflow#43120]
I am trying to fit my model by using a generator working on my dataset (audio files). I use a gold cluster (SLURM) to run my program including generator for creating my input features and corresponding targets (by using yeilding method, and safe threading), and fit_generator for training my model.
Unfortunetly, after 388 iterations including batches with 128 size, the error:

 tensorflow.python.framework.errors_impl.InvalidArgumentError:  input and filter must have the same depth: 6 vs 18
	 [[node model_4/conv2d_80/BiasAdd (defined at /Project1/Training/Prog/preANDtrain_generator_model2.py:187) ]] [Op:__inference_train_function_9095]

Function call stack:
train_function

2020-09-10 17:09:40.560724: W tensorflow/core/kernels/data/generator_dataset_op.cc:103] Error occurred when finalizing GeneratorDataset iterator: Failed precondition: Python interpreter state is not initialized. The process may be terminated.
	 [[{{node PyFunc}}]] 

was shown and my program was failed. 
The depth (#channels) of the input features is 6 and the first layer is a covolutional layer with 18 channels (filters). I became confused why the error was shown after 388 iterations, and why the depth of input features and the first layer must be same.
I will be so grateful if someone can help me to solve the problem
[New Text Document.txt](https://github.com/tensorflow/tensorflow/files/5204020/New.Text.Document.txt)
.

**Comment by PEYMAN333** [comment]
Thank you so much bhack for your answer.

I have just tried **fit** instead of **fit_generator** in my program, and the same error was shown at the same iteration (388).
It is really confusing that the error arises after several iterations.

I created a template for this issue and attached a very very minimal program and dataset: [relatedtoissue_43120](https://github.com/PEYMAN333/relatedtoissue_43120) https://github.com/PEYMAN333/relatedtoissue_43120

Thank you again for your help.

**Comment by PEYMAN333** [comment]
I performed line to line double checks, and found out that one line has unwanted TAB, in one of the functions of the generator. This could lead to a reduced dimension feature in audio files with small size. I edited and run again the program. Fortunately, the training phase passed the 388 iteration. However, I am still surprised for the strange shown error which may seem irrelevant! The "ValueError: Error when checking input: expected input_ to have * dimensions, but got array with shape (, ,)**" may be more relevant for this dimension mismatching.
